flows/engine/session_test.go

```diff
@@ -77,6 +80,53 @@ func TestEvaluateTemplateAsString(t *testing.T) {
 		{"@twitter_handle", "@twitter_handle", false},
 	}
 
+	run := createTestSession(t)
+
+	for _, test := range tests {
+		eval, err := run.EvaluateTemplateAsString(test.template, false)
+		if test.hasError {
+			assert.Error(t, err, "expected error evaluating template '%s'", test.template)
+		} else {
+			assert.NoError(t, err, "unexpected error evaluating template '%s'", test.template)
+			assert.Equal(t, test.expected, eval, "Actual '%s' does not match expected '%s' evaluating template: '%s'", eval, test.expected, test.template)
+		}
+	}
+}
+
+func TestContextToJSON(t *testing.T) {
+	tests := []struct {
+		path     string
+		expected string
+	}{
+		{"contact.uuid", `"ba96bf7f-bc2a-4873-a7c7-254d1927c4e3"`},
+		{"contact.name", `"Ben Haggerty"`},
+		{"contact.urns", `[{"display":"","path":"+12065551212","scheme":"tel"},{"display":"","path":"1122334455667788","scheme":"facebook"},{"display":"","path":"ben@macklemore","scheme":"mailto"}]`},
+		{"contact.urns.0", `{"display":"","path":"+12065551212","scheme":"tel"}`},
+		{"contact.fields", `{"activation_token":"","age":23,"first_name":"Bob","gender":"","joined":"2018-03-27T10:30:00.123456+02:00","state":"Azuay"}`},
+		{"contact.fields.age", `23`},
+		{"contact", `{"channel":{"address":"+12345671111","name":"Nexmo","uuid":"57f1078f-88aa-46f4-a59a-948a5739c03d"},"fields":{"activation_token":"","age":23,"first_name":"Bob","gender":"","joined":"2018-03-27T10:30:00.123456+02:00","state":"Azuay"},"groups":[{"name":"Azuay State","uuid":"d7ff4872-9238-452f-9d38-2f558fea89e0"},{"name":"Survey Audience","uuid":"2aad21f6-30b7-42c5-bd7f-1b720c154817"}],"language":"eng","name":"Ben Haggerty","timezone":"America/Guayaquil","urns":[{"display":"","path":"+12065551212","scheme":"tel"},{"display":"","path":"1122334455667788","scheme":"facebook"},{"display":"","path":"ben@macklemore","scheme":"mailto"}],"uuid":"ba96bf7f-bc2a-4873-a7c7-254d1927c4e3"}`},
+		{"run.input", `{"attachments":[{"content_type":"image/jpeg","url":"http://s3.amazon.com/bucket/test_en.jpg?a=Azuay"}],"channel":{"address":"+12345671111","name":"Nexmo","uuid":"57f1078f-88aa-46f4-a59a-948a5739c03d"},"created_on":"2000-01-01T00:00:00.000000Z","text":"Hi there","type":"msg","urn":{"display":"","path":"+12065551212","scheme":"tel"},"uuid":"84f8a3cf-0f2c-4881-9502-2d7b114bf01f"}`},
+
+		// TODO add way to mock call calls to Now() so we can have deterministic tests without doing text substitution of dates?
```

Comment:
this would be nice too for the tests for flowrunner and not so different to how we mock new UUID generation